diff -urNp -x '*.orig' hplip-3.22.6.org/Makefile.am hplip-3.22.6/Makefile.am
--- hplip-3.22.6.org/Makefile.am	2022-07-10 22:42:55.852246806 +0200
+++ hplip-3.22.6/Makefile.am	2022-07-10 22:42:59.782176747 +0200
@@ -71,7 +71,7 @@ else
 libsane_hpaio_la_LDFLAGS = -version-info 1:0:0
 endif
 # The following is a interlibrary dependency that must be compiled first.
-libsane_hpaio_la_LIBADD = libhpip.la libhpmud.la libhpipp.la $(DBUS_LIBS) -lcups -ldl
+libsane_hpaio_la_LIBADD = libhpip.la libhpmud.la libhpipp.la $(DBUS_LIBS) -lcups -ldl -lc
 #libsane_hpaio_la_CFLAGS = -DWITH_NONAMESPACES -DSOAP_DEBUG
 libsane_hpaio_la_CFLAGS = $(DBUS_CFLAGS) -Iprotocol
 
@@ -324,7 +324,7 @@ lib_LTLIBRARIES += libhpipp.la
 libhpipp_la_SOURCES = protocol/hp_ipp.c protocol/hp_ipp.h protocol/hp_ipp_i.h
 libhpipp_la_CFLAGS = -DCONFDIR=\"$(hplip_confdir)\"
 libhpipp_la_LDFLAGS = -version-info 0:1:0
-libhpipp_la_LIBADD = libhpmud.la
+libhpipp_la_LIBADD = libhpmud.la -lcups
 
 # cupsext
 cupsextdir = $(pyexecdir)
diff -urNp -x '*.orig' hplip-3.22.6.org/scan/sane/ledm.c hplip-3.22.6/scan/sane/ledm.c
--- hplip-3.22.6.org/scan/sane/ledm.c	2022-06-23 06:56:30.000000000 +0200
+++ hplip-3.22.6/scan/sane/ledm.c	2022-07-10 22:42:59.782176747 +0200
@@ -1078,7 +1078,7 @@ void ledm_close(SANE_Handle handle)
 
   if (ps == NULL || ps != session)
   {
-    BUG("invalid sane_close\n");
+//    BUG("invalid sane_close\n");
     return;
   }
 
