--- hplip-3.10.9/scan/sane/ledm.c.wiget	2010-11-07 18:01:26.887129289 +0100
+++ hplip-3.10.9/scan/sane/ledm.c	2010-11-07 18:02:17.662175777 +0100
@@ -973,7 +973,7 @@ void ledm_close(SANE_Handle handle)
 
   if (ps == NULL || ps != session)
   {
-    BUG("invalid sane_close\n");
+//    BUG("invalid sane_close\n");
     return;
   }
 
--- hplip-3.10.9/Makefile.am.wiget	2010-11-07 18:02:32.254945320 +0100
+++ hplip-3.10.9/Makefile.am	2010-11-07 18:03:03.497071268 +0100
@@ -56,7 +56,7 @@ else
 libsane_hpaio_la_LDFLAGS = -version-info 1:0:0
 endif
 # The following is a interlibrary dependency that must be compiled first.
-libsane_hpaio_la_LIBADD = libhpip.la libhpmud.la $(DBUS_LIBS) -lcups -ldl
+libsane_hpaio_la_LIBADD = libhpip.la libhpmud.la $(DBUS_LIBS) -lcups -ldl -lc
 #libsane_hpaio_la_CFLAGS = -DWITH_NONAMESPACES -DSOAP_DEBUG
 libsane_hpaio_la_CFLAGS = $(DBUS_CFLAGS)
 endif # SCAN_BUILD
